pubtools/pulplib/_impl/client/client.py

```diff
@@ -179,6 +182,42 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def upload_one_file(
+        self,
+        file_name,
+        repo_id,
+        type_id,
+        unit_key,
+        unit_metadata=None,
+        override_config=None,
+    ):
+        """Including four steps uploading a file:
+            1. request upload id from pulp
+            2. upload contents to pulp
+            3. import the uploaded content to wanted repo
+            4. delete the reuqest id
+        """
+        upload_id = self._request_upload().result()["upload_id"]
+        offset = 0
+        with open(file_name) as f:
+            data = f.read(self._PULP_CHUNK_SIZE)
+            LOG.info("Uploading %s to repo %s", file_name, repo_id)
+            while data:
+                self._do_upload(data, upload_id, offset).result()
+                offset += self._PULP_CHUNK_SIZE
+                data = f.read(self._PULP_CHUNK_SIZE)
+
+        self._do_import(
+            repo_id,
+            upload_id,
+            unit_type_id=type_id,
+            unit_key=unit_key,
+            unit_metadata=unit_metadata,
+            override_config=override_config,
+        ).result()
+
+        self._delete_upload_request(upload_id).result()
```

Comment:
I would rather suggest just fire-and-forget for deleting the upload request.  The caller doesn't actually need the upload request to be deleted before proceeding, and also if the deletion fails, the upload hasn't really failed.

In other words I think it's fine to just write:

```
# Queue deletion of upload request, but we don't wait for that to succeed
self._delete_upload_request(upload_id)
```

delete_upload_request probably should just log a warning if it fails though.